patches/server/0979-Allow-LEFT_CLICK_AIR-in-3.0-4.5-range-of-entity.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Moulberry <james.jenour@protonmail.com>
+Date: Sun, 14 May 2023 21:44:40 +0800
+Subject: [PATCH] Allow LEFT_CLICK_AIR in 3.0->4.5 range of entity
+
+In Creative, the block and entity reach is the same (4.5), so the code
+works as is.
+In Adventure, the event always fires, so the code works as is.
+In Survival, the block reach is 4.5 and the entity reach is 3.0.
+Therefore, the existing code will not fire a LEFT_CLICK_AIR if an entity
+is between 3.0 and 4.5 blocks away.
+
+To be honest, this entire method of performing a raycast to decide
+whether or not to fire a LEFT_CLICK_AIR is rather error-prone, even with
+this fix. As of 1.19.4, the vanilla client will send the block/entity
+interaction packets before the swing, so using that information to
+decide whether or not to fire LEFT_CLICK_AIR would (in my opinion) be
+more reliable.
+
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index aa287d7f37f38d938d195114408cb6dbda59063d..6bff5bb0c80e6ac55fbb207a5cfa356f5395278f 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -2613,7 +2613,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
+         // SPIGOT-5607: Only call interact event if no block or entity is being clicked. Use bukkit ray trace method, because it handles blocks and entities at the same time
+         org.bukkit.util.RayTraceResult result = this.player.level.getWorld().rayTrace(origin, origin.getDirection(), d3, org.bukkit.FluidCollisionMode.NEVER, false, 0.1, entity -> entity != this.player.getBukkitEntity() && this.player.getBukkitEntity().canSee(entity));
+ 
+-        if (result == null || this.player.gameMode.getGameModeForPlayer() == GameType.ADVENTURE) { // Paper - call PlayerInteractEvent when left-clicking on a block in adventure mode
++        if (result == null || this.player.gameMode.getGameModeForPlayer() == GameType.ADVENTURE || (player.gameMode.getGameModeForPlayer() != GameType.CREATIVE && result.getHitEntity() != null && origin.toVector().distanceSquared(result.getHitPosition()) > 3.0D*3.0D)) { // Paper - call PlayerInteractEvent when left-clicking on a block in adventure mode // Paper - call PlayerInteractEvent when entity distance is greater than 3.0
```

Comment:
Double-paper comments for this are a bit iffy. Maybe instead we could just squash this into the original patch, and make the logic a bit more readable here.